### ISSUE TITLE
Model with morphOne doesn't update morphable entity on insertOrUpdate

### ISSUE THREAD

**Issue by RickMeijer** [vuex-orm/vuex-orm#218]
Hi,

First off: amazing project you're running, KiaKing. Super impressed. I hope I can contribute to it at some point in the future.  For now I'm afraid I'm to much of a newbie. So let's contribute in the form of a bug report! Maybe I'm doing something wrong, but I can't see it anymore.

I'm having trouble updating existing data with a morphOne/polymorphic relationship.
In the demo below there is a Trip model and a Geodata model.
The Trip has a morphOne relationship to the Geodata.
When you click the button a new trip is `insertOrUpdate`, resulting in one trip entity, and one geodata entity.

When I click again to update the trip, the trip gets updated, but another geodata entry is made.
I've made a small demo where you can see the effect:

[![Edit Vue Template](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/326p4pxv7p)

### What I expect:
First click:
- Creates trip
- Creates geodata
- Trip.geodata is not null

Second Click:
- Updates Trip
- Updates GeoData

### What actually happens:
First click:
- Creates trip
- Trip.geodata is null
- Creates geodata

Second Click:
- Updates Trip
- New GeoData

**Comment by kiaking** [comment]
Hi! Thank you so much for your kind word and report! Those are the things makes me motivated : )

OK, the reason why it is creating a new record is because you don't have `id` field at `GeoData`. If you add `id` field to the GeoData record when inserting it, it will be updated. If it's hard for you to add single primary key to the GeoData, you could also define the composite primary key.

```js
class GeoData extends Model {
  static entity = "geodatas";

  static primaryKey = ['geodatable_id', 'geodatable_type'];

  static fields() {
    return {
      lat: this.number(null),
      lng: this.number(null),
      geodatable_id: this.number(null),
      geodatable_type: this.attr(null),
      geodatable: this.morphTo("geodatable_id", "geodatable_type")
    };
  }
}
```

Here is the working example.

[![Edit Vue Template](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/9o57q713my)

**Comment by RickMeijer** [comment]
> OK, the reason why it is creating a new record is because you don't have id field at GeoData
Yeah; that's what I figured. But since it has a morphOne relationship I also figured that an ID should not be neccesary, since there's always one relation only.  
I did not realize you could composite primary keys like that! That's supernice. 
Time for some refactoring on other objects lol.

There's only one issue remaining; when I multiple items as an array only one relation gets inserted.
I've modified your example below to show you what I mean. 

I've added a method `trip` that generates a record with either a random ID (handy to test insert/update) or with an argument. This method is called for the dispatchAction to send the data.
I expect for every inserted item to have a relation item as well.
However; this is not the case. Is this intentional?

[![Edit Vue Template](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/9384qjryor)

**Comment by kiaking** [comment]
Oh, I think this is a defect. In this case it should indeed create 2 geodatas. Thank you for the report!

I'll look into this and make a fix. Meanwhile you could workaround the issue by inserting records one by one, if that suits your needs...

**Comment by RickMeijer** [comment]
I've written a sort-of plugin that timestamps the insertion/creation records that kinda slows things down right now since theres a boatload of mutations now, but I'll manage for now, thanks!

Oh, and using a composite primary key to fix non-updating morphOne relations doesn't scale, unfortunately. 
I've also got a morphMany relationship from another model to geodata, so I can't use type and id since that would convery any Many to a single item that keeps being overwritten, if that makes sense. I'll figure something out though, thanks again! 

Again: you're doing crazy good work. Thank you so much for time spent already.

**Comment by kiaking** [comment]
> I've also got a morphMany relationship from another model to geodata, so I can't use type and id since that would convery any Many to a single item that keeps being overwritten, if that makes sense. For now I'm adding it to the data before committing it to the store instead of relying on the composite primary key.

Ah, I see. It's kinda hard to use `insertOrUpdate` method without a unique primary key because it's tough to determine if the record is new or existing one. Maybe there's some kind of solution...? I'll think about this...!

> Again: you're doing crazy good work. Thank you so much for time spent already.

Thank you so much! There're a lot more that I want to do with this library. Need to make more time :rocket:

**Comment by RickMeijer** [comment]
Sorry for the late reply.
I eventually found a way to work around it apparently, but I can't seem to remember what exactly it is I did. Two months ago the project looked a lot different than it does now. I'm not even sure if this issue is still relevant for it, to be completely honest.

Right now the main issue I'm dealing with is that updating (db/insert) one item triggers a refresh on all the items. This kind of sucks when doing the realtime vehicle tracking the app needs to do. Updating the position of one vehicle recalculates the entire tree of vehicles and relevant data and relations.

I've no idea how Vuex ORM should behave; it's a bit beyond my skill level. When I'd do it myself I'd probably do something like (pseudocode). 

```
db[entity][data.id] = {...db[entity][data.id], ...data }
```

Sorry I can't be of more help.
